packages/snaps-sdk/src/internals/jsx.ts

```diff
@@ -34,11 +35,13 @@ type StructSchema<Type> =
       : [Type] extends [string | undefined | null]
         ? [Type] extends [`0x${string}`]
           ? null
-          : [Type] extends [IsMatch<Type, string | undefined | null>]
+          : [Type] extends [CaipChainId]
```

Comment:
Why is this needed?

Reply:
See convo here: https://github.com/MetaMask/snaps/pull/3129#discussion_r1973489336